src/plugins/presentation_util/public/components/input_controls/control_types/options_list/options_list_component.tsx

```diff
@@ -6,133 +6,67 @@
-import useMount from 'react-use/lib/useMount';
+import React, { useState } from 'react';
 import classNames from 'classnames';
-import { Subject } from 'rxjs';
-import { EuiFilterButton, EuiFilterGroup, EuiPopover, EuiSelectableOption } from '@elastic/eui';
 
-import {
-  OptionsListDataFetcher,
-  OptionsListEmbeddable,
-  OptionsListEmbeddableInput,
-} from './options_list_embeddable';
+import { EuiFilterButton, EuiFilterGroup, EuiPopover, EuiSelectableOption } from '@elastic/eui';
+import { Subject } from 'rxjs';
+import useMount from 'react-use/lib/useMount';
 import { OptionsListStrings } from './options_list_strings';
-import { InputControlOutput } from '../../embeddable/types';
 import { OptionsListPopover } from './options_list_popover_component';
-import { withEmbeddableSubscription } from '../../../../../../embeddable/public';
 
 import './options_list.scss';
 
-const toggleAvailableOptions = (
-  indices: number[],
-  availableOptions: EuiSelectableOption[],
-  enabled: boolean
-) => {
-  const newAvailableOptions = [...availableOptions];
-  indices.forEach((index) => (newAvailableOptions[index].checked = enabled ? 'on' : undefined));
-  return newAvailableOptions;
-};
-
-interface OptionsListProps {
-  input: OptionsListEmbeddableInput;
-  fetchData: OptionsListDataFetcher;
+export interface OptionsListComponentState {
+  availableOptions?: EuiSelectableOption[];
+  selectedOptionsString?: string;
+  selectedOptionsCount?: number;
+  twoLineLayout?: boolean;
+  searchString?: string;
+  loading: boolean;
 }
 
-export const OptionsListInner = ({ input, fetchData }: OptionsListProps) => {
-  const [availableOptions, setAvailableOptions] = useState<EuiSelectableOption[]>([]);
-  const selectedOptions = useRef<Set<string>>(new Set<string>());
-
-  // raw search string is stored here so it is remembered when popover is closed.
-  const [searchString, setSearchString] = useState<string>('');
-  const [debouncedSearchString, setDebouncedSearchString] = useState<string>();
-
+export const OptionsListComponent = ({
+  componentStateSubject,
+  typeaheadSubject,
+  updateOption,
+}: {
+  componentStateSubject: Subject<OptionsListComponentState>;
+  typeaheadSubject: Subject<string>;
+  updateOption: (index: number) => void;
+}) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
-  const [loading, setIsLoading] = useState(false);
-
-  const typeaheadSubject = useMemo(() => new Subject<string>(), []);
+  const [optionsListState, setOptionsListState] = useState<OptionsListComponentState>(
+    {} as OptionsListComponentState
+  );
 
   useMount(() => {
-    typeaheadSubject
-      .pipe(
-        tap((rawSearchText) => setSearchString(rawSearchText)),
-        debounceTime(100)
-      )
-      .subscribe((search) => setDebouncedSearchString(search));
-    // default selections can be applied here...
+    componentStateSubject.subscribe((newState) => setOptionsListState(newState));
```

Comment:
For the mount vs effect discussion, I think if the componentStateSubject were to change, then I think this component would stop working as it wouldn't subscribe to the new subject.  I don't know if we'll ever hit that scenario though, but I think it is a potential bug.

Reply:
In this specific example, it was okay for useMount to be used, because the only way for the subscription to be changed is for the embeddable to be destroyed and reconstructed, which would mount a new component on render. 

That said, I've made this into a generic hook `use_state_observable` so that we can use a similar pattern in other input controls. Because it's now generic, a `useEffect` makes much more sense. Additionally, I have cleaned up the subscription in the useEffect cleanup. 

Great suggestion!